raster/src/main/scala/geotrellis/raster/reproject/RasterRegionReproject.scala

```diff
@@ -21,40 +21,162 @@ trait RasterRegionReproject[T <: CellGrid] extends Serializable {
+
+              (pxarr, xres, yres)
+            case p: Point =>
+              val (x, y) = toSrcCrs(p.x, p.y)
+              (Array(destRasterExtent.mapXToGrid(p.x)), Array(x), Array(y))
+            case g: Geometry =>
+              throw new IllegalStateException("Line-Polygon intersection cannot produce geometries that are not Points or Lines")
+            }
+        }).fold( (Array.empty[Int], Array.empty[Double], Array.empty[Double]) ){ (a1, a2) => {
+          val (px1, x1, y1) = a1
+          val (px2, x2, y2) = a2
+          (px1 ++ px2, x1 ++ x2, y1 ++ y2)
+        }}
+      } else
+        (Array.empty[Int], Array.empty[Double], Array.empty[Double])
+    }
+  }
+
   implicit val singlebandInstance = new RasterRegionReproject[Tile] {
     def regionReproject(raster: Raster[Tile], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod): Raster[Tile] = {
       val buffer = raster.tile.prototype(rasterExtent.cols, rasterExtent.rows)
       val trans = Proj4Transform(dest, src)
       val resampler = Resample.apply(resampleMethod, raster.tile, raster.extent, CellSize(raster.rasterExtent.cellwidth, raster.rasterExtent.cellheight))
 
+      val rowcoords = rowCoords(region, rasterExtent, trans)
+
+      cfor(0)(_ < rasterExtent.rows, _ + 1){ i =>
+        val (pxs, xs, ys) = rowcoords(i)
+        if (raster.cellType.isFloatingPoint) {
+          cfor(0)(_ < xs.size, _ + 1){ s =>
+            buffer.setDouble(pxs(s), i, resampler.resampleDouble(xs(s), ys(s)))
+          }
+        } else {
+          cfor(0)(_ < xs.size, _ + 1){ s =>
+            buffer.set(pxs(s), i, resampler.resample(xs(s), ys(s)))
+          }
+        }
+      }
+
+      Raster(buffer, rasterExtent.extent)
+    }
+
+    def mutableRegionReproject(target: Tile, raster: Raster[Tile], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod) = {
+      val buffer = target.mutable
+      val trans = Proj4Transform(dest, src)
+      val resampler = Resample.apply(resampleMethod, raster.tile, raster.extent, CellSize(raster.rasterExtent.cellwidth, raster.rasterExtent.cellheight))
+
+      val rowcoords = rowCoords(region, rasterExtent, trans)
+
+      cfor(0)(_ < rasterExtent.rows, _ + 1){ i =>
+        val (pxs, xs, ys) = rowcoords(i)
+        if (raster.cellType.isFloatingPoint) {
+          cfor(0)(_ < xs.size, _ + 1){ s =>
+            buffer.setDouble(pxs(s), i, resampler.resampleDouble(xs(s), ys(s)))
+          }
+        } else {
+          cfor(0)(_ < xs.size, _ + 1){ s =>
+            buffer.set(pxs(s), i, resampler.resample(xs(s), ys(s)))
+          }
+        }
+      }
+    }
+  }
+
+  implicit val multibandInstance = new RasterRegionReproject[MultibandTile] {
+    def regionReproject(raster: Raster[MultibandTile], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod): Raster[MultibandTile] = {
```

Comment:
The code duplication between `regionReproject` and `mutableRegionReproejct` should be taken care of. Probably the easiest thing is to delegate from the former to the latter once tiles are allocated.

Reply:
Any thoughts on naming convention for `mutable` methods now that its used?

Reply:
I don't have significant thoughts on naming convention, sad to say.  This works well enough for me.  Do you have specific concerns about the nomenclature used here?